contracts/liquidator/Liquidator.sol

```diff
@@ -145,37 +250,41 @@ contract Liquidator is IUniswapV3FlashCallback, PeripheryImmutableState, Periphe
             address asset = comet.getAssetInfo(i).asset;
             cometAssets[i] = asset;
             uint256 collateralBalance = comet.collateralBalanceOf(address(comet), asset);
-            uint256 quotePrice = comet.quoteCollateral(asset, 1 * comet.baseScale());
-
-            /*
-                quoteCollateral = amount of DAI you get for 1 USDC
-                collateralBalance = Comet's balance of DAI
-                price = amount of USDC required to buy all DAI
-
-                1 / quotePrice = x / collateralBalance
-                (1 / quotePrice) * collateralBalance = x
-            */
-            // uint256 assetBaseAmount = comet.collateralBalanceOf(address(comet), asset) * quotePrice / 1e30; // PRICE_SCALE + 1e12
-            // console.log("assetBaseAmount: %s", assetBaseAmount);
-            uint256 assetBaseAmount = ((1e6 * 1e18 / quotePrice) * collateralBalance) / 1e18;
+
+            if (collateralBalance == 0) continue;
+
+            // Find the price in asset needed to base QUOTE_PRICE_SCALE of USDC(base token) of collateral
+            uint256 quotePrice = comet.quoteCollateral(asset, QUOTE_PRICE_SCALE * comet.baseScale());
+            uint256 assetBaseAmount = comet.baseScale() * QUOTE_PRICE_SCALE * collateralBalance / quotePrice;
+
+            // Liquidate only positions with adequate gains, no need to collect residue from protocol
```

Comment:
nit: this actually checks the size of the liquidation, but not the gains

Reply:
comment updated in #474